src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -3133,6 +3222,76 @@ private static bool DeclareSameIdentifiers(SeparatedSyntaxList<VariableDeclarato
             return true;
         }
 
+        internal override void ReportSemanticRudeEdits(SemanticModel oldModel, SyntaxNode oldNode, SemanticModel newModel, SyntaxNode newNode, List<RudeEditDiagnostic> diagnostics)
+        {
+            var foundNode = FindUnsupportedV7Switch(oldModel, oldNode, diagnostics);
+            if (foundNode != null)
+            {
+                AddRudeUpdateInCSharp7Method(diagnostics, foundNode);
+            }
+            else if ((foundNode = FindUnsupportedV7Switch(newModel, newNode, diagnostics)) != null)
+            {
+                AddRudeUpdateAroundActiveStatement(diagnostics, foundNode);
+            }
+        }
+
+        private SyntaxNode FindUnsupportedV7Switch(SemanticModel model, SyntaxNode syntaxNode, List<RudeEditDiagnostic> diagnostics)
+        {
+            foreach (var node in syntaxNode.DescendantNodesAndSelf().Where(n => n.Kind() == SyntaxKind.SwitchStatement))
+            {
+                var switchExpression = ((SwitchStatementSyntax)node).Expression;
+                ITypeSymbol governingType = model.GetTypeInfo(switchExpression).Type;
+
+                if (!IsValidV6SwitchGoverningType(governingType))
+                {
+                    return node;
+                }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Returns true iff the supplied type is sbyte, byte, short, ushort, int, uint,
+        /// long, ulong, bool, char, string, or an enum-type, or if it is the nullable type
+        /// corresponding to one of those types. These types were permitted as the governing
+        /// type of a switch statement in C# 6.
+        /// </summary>
+        private static bool IsValidV6SwitchGoverningType(ITypeSymbol type)
```

Comment:
It does, but is internal to the compiler and there was a decision not to make it public for a single consumer.
I've tried to make it visible with InternalsVisibleTo, but it created many problems, so I assume there is a reason why it is not already done. Also I noticed other types and methods in this project which seem to duplicate code from the compiler, which reinforced my assumption.